File: Java/dependency-injection/changing-behavior-through-composition/src/main/java/com/github/grzesiek_galezowski/ChangingBehaviorThroughComposition/Building.java
package com.github.grzesiek_galezowski.ChangingBehaviorThroughComposition;

public class Building {
  private final Alarm _alarm;

  public Building(Alarm alarm) {
    _alarm = alarm;
  }

  public void someoneCameIn() {
    _alarm.trigger();
  }

  public void situationIsClear() {
    _alarm.disable();
  }
}

File: CSharp/DependencyInjection/ChangingBehaviorThroughComposition/Building.cs
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace ChangingBehaviorThroughComposition
{
    public class Building
    {
      private readonly Alarm _alarm;

      public Building(Alarm alarm)
      {
        _alarm = alarm;
      }

      public void SomeoneCameIn()
      {
        _alarm.Trigger();
      }

      public void SituationIsClear()
      {
        _alarm.Disable();
      }
    }
}
